### PR TITLE
Java client accepts topologies with `DEAD` partitions

### LINKED ISSUE #9387: Java client fails to describe partitions that are `DEAD`
**Describe the bug**

`PartitionInfoImpl` only accepts partitions that are `HEALTHY` or `UNHEALTHY` and throws an exception for partitions which are `DEAD`. 

https://github.com/camunda/zeebe/blob/1edfe4f762ef700885859bf87e1370963d3448a1/clients/java/src/main/java/io/camunda/zeebe/client/impl/response/PartitionInfoImpl.java#L46-L56

**Expected behavior**

`PartitionInfoImpl` should  also support the `DEAD` status and not throw an exception.

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

The Java client now accepts partitions that are `DEAD` instead of throwing an exception. I've also added a test that generates randomized topology responses and ensures that they are accepted by the Java client.

closes #9387

### CODE REVIEW

**Comment by oleschoenburg** [comment]
@saig0 Since you reacted to the original issue, are you interested in reviewing this? :wink:

**Comment by oleschoenburg** [comment]
Hm, looks like I can't get Junit 5 to work for the java-client module :thinking: 
```
[2022-05-19T06:14:48.670Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M6:test (default-test) on project zeebe-client-java: 
[2022-05-19T06:14:48.670Z] [ERROR] 
[2022-05-19T06:14:48.670Z] [ERROR] Please refer to /home/jenkins/agent/workspace/9387-java-client-dead-partitions/clients/java/target/surefire-reports for the individual test results.
[2022-05-19T06:14:48.670Z] [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[2022-05-19T06:14:48.670Z] [ERROR] There was an error in the forked process
[2022-05-19T06:14:48.670Z] [ERROR] TestEngine with ID 'junit-jupiter' failed to discover tests
[2022-05-19T06:14:48.670Z] [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[2022-05-19T06:14:48.670Z] [ERROR] TestEngine with ID 'junit-jupiter' failed to discover tests
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:699)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:311)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:268)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1332)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1165)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:929)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:196)
[2022-05-19T06:14:48.670Z] [ERROR] 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:186)
[2022-05-19T06:14:48.671Z] [ERROR] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[2022-05-19T06:14:48.671Z] [ERROR] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[2022-05-19T06:14:48.671Z] [ERROR] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[2022-05-19T06:14:48.671Z] [ERROR] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2022-05-19T06:14:48.671Z] [ERROR] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2022-05-19T06:14:48.671Z] [ERROR] 	at java.lang.Thread.run(Thread.java:750)
```

File: clients/java/pom.xml [hunk 1]
```diff
@@ -139,6 +139,21 @@
...
+      <!-- Explicitely depend on 4.3 since newer versions require Java 11 -->
+      <version>4.3.0</version>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>io.github.murdos</groupId>
+      <artifactId>easy-random-protobuf</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (saig0) [inline clients/java/pom.xml:155]: :x: The versions of the dependencies are not compatible.

```
Dependency convergence error for org.jeasy:easy-random-core:jar:4.3.0:test paths to dependency are:
Failed while enforcing releasability. See above detailed error message.
+-io.camunda:zeebe-client-java:jar:8.1.0-SNAPSHOT
  +-org.jeasy:easy-random-core:jar:4.3.0:test
and
+-io.camunda:zeebe-client-java:jar:8.1.0-SNAPSHOT
  +-io.github.murdos:easy-random-protobuf:jar:0.4.0:test
    +-org.jeasy:easy-random-core:jar:5.0.0:test

Warning:  Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
```

**Review by saig0 — CHANGES_REQUESTED** [review]
@oleschoenburg thank you for providing a fix :+1: 

I have suggestion about the test. Please have a look.

File: clients/java/src/test/java/io/camunda/zeebe/client/impl/response/TopologyImplTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    final Stream<TopologyResponse> responses = random.objects(TopologyResponse.class, 100);
+
+    // then
+    responses.forEach(
+        topologyResponse -> {
+          assertThatCode(() -> new TopologyImpl(topologyResponse))
+              .withFailMessage("Converting response caused an exception: <%s>", topologyResponse)
+              .doesNotThrowAnyException();
+        });
+  }
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply clients/java/src/test/java/io/camunda/zeebe/client/impl/response/TopologyImplTest.java:48]: Yeah that makes sense to me :+1: The nice thing about this randomized test would have been that it doesn't just test the handling of the health enum but the entire topology response.

**Review by saig0 — APPROVED** [review]
@oleschoenburg LGTM :+1:

**Comment by saig0** [comment]
bors merge

**Comment by saig0** [comment]
@korthout something is wrong with the backport action :sweat_smile: can you have a look?

**Comment by korthout** [comment]
Thanks for letting me know. It seems to be related to the request review feature. To request reviews, the @backport-action user must have admin rights to this repo, which it doesn't have. I'll open a bug for it. 

As a workaround in zeebe, until a fix is available, we could give the backport-action user admin rights to the repo. For example, by adding it to the [Zeebe team](https://github.com/orgs/camunda/teams/zeebe/members). However, I don't think we should because the account is not controlled by Camunda, and therefore less secured. Luckily it seems to open PRs correctly.

I think the only solution is to disable the feature or make enabling it configurable. However, that will require a new release.

**Comment by korthout** [comment]
A new version is released https://github.com/zeebe-io/backport-action/releases/tag/v0.0.8

**Comment by korthout** [comment]
Haha wow, it picked up the url as a backport comment. You can ignore these last 2 comments.

**Pull request merged**